tests/reg_tests/test_actuator.py

```diff
@@ -0,0 +1,814 @@
+            "mgcycle": "sg",
+            "ncycles": 1000,
+            "useanksolver": True,
+            "usenksolver": True,
+            "anksecondordswitchtol": 1e-2,
+            "nkswitchtol": 1e-4,
+            "volumevariables": ["temp", "mach", "resrho"],
+            "surfacevariables": ["temp", "vx", "vy", "vz", "p", "ptloss", "mach", "rho"],
+            "equationType": "Euler",
+            "l2convergence": 1e-13,
+            "adjointl2convergence": 1e-13,
+        }
+
+        options = copy.copy(adflowDefOpts)
+        options["outputdirectory"] = os.path.join(baseDir, options["outputdirectory"])
+        options.update(self.options)
+
+        CFDSolver = ADFLOW(options=options)
+
+        CFDSolver.addFunction("mdot", "inlet", name="mdot_in")
+        CFDSolver.addFunction("mdot", "outlet", name="mdot_out")
+
+        CFDSolver.addFunction("aavgptot", "outlet", name="aavgptot_out")
+        CFDSolver.addFunction("aavgptot", "inlet", name="aavgptot_in")
+
+        CFDSolver.addFunction("mavgttot", "outlet", name="mavgttot_out")
+        CFDSolver.addFunction("mavgttot", "inlet", name="mavgttot_in")
+
+        CFDSolver.addFunction("aavgps", "outlet", name="aavgps_out")
+        CFDSolver.addFunction("aavgps", "inlet", name="aavgps_in")
+
+        CFDSolver.addFunction("area", "inlet", name="area_in")
+        CFDSolver.addFunction("area", "outlet", name="area_out")
+
+        CFDSolver.addFunction("mavgvx", "inlet", name="mavgvx_in")
+        CFDSolver.addFunction("mavgvx", "outlet", name="mavgvx_out")
+
+        CFDSolver.addFunction("forcexpressure", "inlet", name="forcexpressure_in")
+        CFDSolver.addFunction("forcexpressure", "outlet", name="forcexpressure_out")
+
+        CFDSolver.addFunction("forcexmomentum", "inlet", name="forcexmomentum_in")
+        CFDSolver.addFunction("forcexmomentum", "outlet", name="forcexmomentum_out")
+
+        self.CFDSolver = CFDSolver
+
+        # this is imported from reg_aeroproblems utility script
+        self.ap = ap_actuator_pipe
+
+        actuatorFile = os.path.join(baseDir, "../../input_files/actuator_test_disk.xyz")
+        self.CFDSolver.addActuatorRegion(
+            actuatorFile,
+            np.array([0, 0, 0]),
+            np.array([1, 0, 0]),
+            "actuator_region",
+            # we will set these individually in the tests below
+            thrust=0.0,
+            torque=0.0,
+            heat=0.0,
+        )
+
+        # add thrust and heat as AP DVs
+        self.ap.setBCVar("Thrust", 0.0, "actuator_region")
+        self.ap.addDV("Thrust", family="actuator_region", units="N", name="thrust")
+
+        self.ap.setBCVar("Heat", 0.0, "actuator_region")
+        self.ap.addDV("Heat", family="actuator_region", units="J/s", name="heat")
+
+        # also add flowpower as an AZ function
+        CFDSolver.addFunction("flowpower", "actuator_region", name="flowpower_az")
+
+    def test_actuator_momentum_flux(self):
```

Comment:
If it tests both momentum and power we should change the test name

Reply:
So I think I meant by test momentum flux, is to test adding thrust terms, and checking if it has the correct effect on both momentum and energy fluxes through the boundaries (you can add thrust in the actuator zone and it both adds momentum to the flow, and does work on the flow) 

I renamed the tests. Let me know if they are more clear.